test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/DefaultHtmlGeneratorTest.cs

```diff
@@ -60,6 +60,122 @@ public void GetCurrentValues_WithNullExpressionResult_ReturnsNull(bool allowMult
             Assert.Null(result);
         }
 
+        [Fact]
+        public void GetCurrentValues_WithNullExpression_Throws()
+        {
+            // Arrange
+            var metadataProvider = new TestModelMetadataProvider();
+            var htmlGenerator = GetGenerator(metadataProvider);
+            var viewContext = GetViewContext<Model>(model: null, metadataProvider: metadataProvider);
+            var modelExplorer = metadataProvider.GetModelExplorerForType(typeof(string), model: null);
+
+            var expected = string.Format(
+                            "The name of an HTML field cannot be null or empty. Instead " +
+                            "use methods {0}.{1} or {2}.{3} with a non-empty {4} " +
+                            "argument value.\r\nParameter name: expression",
+                            typeof(IHtmlHelper).FullName,
+                            nameof(IHtmlHelper.Editor),
+                            typeof(IHtmlHelper<>).FullName,
+                            nameof(IHtmlHelper<object>.EditorFor),
+                            "htmlFieldName");
```

Comment:
1. Helpful in tests to expand everything so it's possible to see the final message. But at least use string interpolation so readers don't need to figure out which argument goes where.
2. Don't indent more than Visual Studio does by default plus, in a few cases, 4 spaces to offset something nested. This applies to many new tests -- some lines almost seem right-justified :smiley_cat: